fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/AbstractLoanScheduleGenerator.java

```diff
@@ -139,8 +139,13 @@ private LoanScheduleModel generate(final MathContext mc, final LoanApplicationTe
         if (!scheduleParams.isPartialUpdate()) {
             BigDecimal downPaymentAmount = BigDecimal.ZERO;
             if (loanApplicationTerms.isDownPaymentEnabled()) {
-                downPaymentAmount = MathUtil.percentageOf(scheduleParams.getOutstandingBalance().getAmount(),
-                        loanApplicationTerms.getDisbursedAmountPercentageForDownPayment(), 19);
+                double downPaymentAmt = MathUtil.percentageOf(scheduleParams.getOutstandingBalance().getAmount(),
```

Comment:
Please dont cast double from BigDecimal and later cast it back to BigDecimal. It should be done the rounding with BigDecimal.